### PR TITLE
Fix Out of Memory issue when allocating bias buffer

### PR DESCRIPTION
**Pull request by amd-jiali**
…s, the maximum memory usage should be re-calculated and reduce the data size to avoid the Out of Memory issue; if without bias, no need to allocate buffers for bias

## Details
___Do not mention proprietary info or link to internal work items in this PR.___

**Work item:** _"Internal", or link to GitHub issue (if applicable)._

**What were the changes?**  
Add an argument to select testing bias or not and calculate data size based on available maximum memory size; differentiate memory allocations w/ and w/o bias.

**Why were the changes made?**  
Issue was observed in SWDEV-570898.

**How was the outcome achieved?**  
Tested on MI300X and MI308X machines, no out-of-memory issue observed.

**Additional Documentation:**  
_What else should the reviewer know?_

### CODE REVIEW

File: src/common.cu [hunk 1]
```diff
@@ -1189,14 +1192,15 @@ int main(int argc, char* argv[]) {
     {"output_file", required_argument, 0, 'x'},                     //RCCL
     {"output_format", required_argument, 0, 'Z'},                   //RCCL
     {"output_algo_proto_channels", required_argument, 0, 'M'},      //RCCL
+    {"test_bias", required_argument, 0, 'B'},
```
Comment (wenkaidu) [inline src/common.cu]: I think you do not need to add extra option for bias. Currently the only supported collective with bias is all reduce, and it has different executable.

Comment (amd-jiali) [reply src/common.cu]: I see, will modify that.

**Review by wenkaidu — APPROVED** [review]

**Review by alex-breslow-amd — APPROVED** [review]
